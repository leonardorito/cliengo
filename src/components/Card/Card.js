import React, { useState } from "react";
import "./Card.scss";
import SwitchToggle from "../SwitchToggle/SwitchToggle";
import PropTypes from 'prop-types'

const Card = (props) => {
	const {data, changedActive} = props;
  return (
    <div className="Card">
      <div className="card-main-container">
				<img src="/assets/conversacion.svg" className="card-img" alt="Conversacion"/>
        <h5>{data.title}</h5>
        <p>{data.description}</p>
      </div>
      <div className="options">
				<img src="/assets/edit.svg" alt="Edit" className="edit-card" />
        <SwitchToggle
					id={data.id}
          isOn={data.active}
          onColor="#1ABC9C"
          handleToggle={changedActive}
        />
      </div>
    </div>
  );
};

Card.propTypes = {
	data: PropTypes.object.isRequired,
	changedActive: PropTypes.func.isRequired
};

export default Card;
