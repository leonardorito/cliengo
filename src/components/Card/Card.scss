@import "../../variables";

.Card {
  background-color: $white;
	box-shadow: 0px 3px 6px $shadow-default;
	border-radius: 4px;
	cursor: pointer;
  .card-main-container {
		padding: 15px 15px 0 32px;
		position: relative;
		.card-img {
			position: absolute;
			left: 8px;
			top: 17px;
		}
	}
  h5 {
    margin: 0;
  }
  .options {
    height: 36px;
    display: flex;
    align-items: center;
		justify-content: flex-end;
		margin-top: 10px;
		padding: 0 20px 0 15px;
		border-top: 1px solid $disabled-gray;

		.edit-card {
			width: 14px;
			height: 14px;
			margin-right: 13px;
		}
  }
}
