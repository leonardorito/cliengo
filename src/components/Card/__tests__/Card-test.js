import React from "react";
import cardData from "../../../constants/cards-data";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Card from "../Card";

it("Card should render with data", () => {
	const answersList = cardData;
	const handleActive = jest.fn(() => {});
  const tree = mount(
    <Card data={answersList[0]} changedActive={handleActive} />
  );
  expect(toJson(tree)).toMatchSnapshot();
});

it("Card should render correct title", () => {
	const answersList = cardData;
	const handleActive = jest.fn(() => {});
  const tree = mount(
    <Card data={answersList[0]} changedActive={handleActive} />
	);
	const title = tree.find('h5');
  expect(title.text()).toBe(answersList[0].title);
});

it("Card should render correct description", () => {
	const answersList = cardData;
	const handleActive = jest.fn(() => {});
  const tree = mount(
    <Card data={answersList[0]} changedActive={handleActive} />
	);
	const description = tree.find('p');
  expect(description.text()).toBe(answersList[0].description);
});
