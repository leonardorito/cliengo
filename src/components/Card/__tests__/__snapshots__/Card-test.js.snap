// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render with data 1`] = `
<Card
  changedActive={[MockFunction]}
  data={
    Object {
      "active": true,
      "description": "Atendemos de lunes a viernes de 9 a 18 hs y s치bados de 10 a 13hs.",
      "id": 1,
      "title": "Horario de atenci칩n",
    }
  }
>
  <div
    className="Card"
  >
    <div
      className="card-main-container"
    >
      <img
        alt="Conversacion"
        className="card-img"
        src="/assets/conversacion.svg"
      />
      <h5>
        Horario de atenci칩n
      </h5>
      <p>
        Atendemos de lunes a viernes de 9 a 18 hs y s치bados de 10 a 13hs.
      </p>
    </div>
    <div
      className="options"
    >
      <img
        alt="Edit"
        className="edit-card"
        src="/assets/edit.svg"
      />
      <SwitchToggle
        handleToggle={[MockFunction]}
        id={1}
        isOn={true}
        onColor="#1ABC9C"
      >
        <div
          className="SwitchToggle"
        >
          <input
            checked={true}
            className="react-switch-checkbox"
            id="react-switch-1"
            onChange={[MockFunction]}
            style={
              Object {
                "display": "none",
              }
            }
            type="checkbox"
          />
          <label
            className="react-switch-label"
            htmlFor="react-switch-1"
            style={
              Object {
                "background": "#1ABC9C",
              }
            }
          >
            <span
              className="react-switch-button"
            />
          </label>
        </div>
      </SwitchToggle>
    </div>
  </div>
</Card>
`;
